### ISSUE TITLE
Unsync feature

### ISSUE THREAD

**Issue by pedrosimao** [rclone/rclone#1760]
I don't know if a similar Feature Request have already been made, but I would love to see it done.
Unsync Feature would allow us to temporarily sync files from the cloud, and keep on our computers while we need. When we don't need anymore we could just "unsync".
The principle is the same used by "odrive" app, which create local placeholders on files that are unsynced. The problem is that odrive is crazy expensive, and it seems to be the only app that does it. For encryption only paid versions works.

rclone currently has all the functions needed to create a odrive like functionality. I think the biggest change that we would need to do, would be to create an rclone agent, that would be constanly running and looking for modifications. Also we would need to add commands/flags like "lightsync" which would sync placeholders instead of the entire files, and then an "unsync" command.

Do you think this is doable? I would love to engage in the project with someone. I am willing to learn some golang and Swift3 only for the project.

**Comment by ncw** [comment]
Is there some docs, or a walkthough somewhere so I can see how odrive does it?

Might an rclone mount with file caching be sufficient for your needs?

**Comment by pedrosimao** [comment]
They do have some documentation here:
https://docs.odrive.com/docs/manage-disk-space

But I think the best way to understand it is to use it.

Concerning file caching, I don't think it is enough. The advantage of odrive is to be able to keep local copies of files for as long as we need/want, without any kind of internet connection. And we don't need the file anymore we can "unsync".  This is very useful when we organize all our files in a folder and sometimes we want to be working on a certain subdirectory, and I want to make sure I will have access to it even without being connected to the internet.

Basically it allow us to have "Cold" and "Hot" directories. The cold directories will be always in the cloud. but the "Hot" directories (or synced directories) will always have a second copy inside our PCs for offline usage. If we modify the files while offline they will be automatically updated when we connect again.

**Comment by pedrosimao** [comment]
Here is a video showing basic usage of odrive:
https://www.odrive.com/amazon-drive-sync

Btw, I have tested odrive agent with CLI commands and it turns out that it is much faster than rclone. Speacially when we backup lot's of small files. With odrive I have been able to download an entire Mac Osx Time Machine Backup I did on Amazon Drive in only one day, while with rclone it took me many days, because the overhead rclone has. Maybe we have something to learn with them. Unhappily they are not an open source project.
